python/doc/examples/functional_modeling/vectorial_functions/plot_functions_inputDim.py

```diff
@@ -0,0 +1,132 @@
+"""
+==========================================
+Increase the input dimension of a function
+==========================================
+"""
+# %%
+# Import the required modules
+
+import openturns as ot
+
+# %%
+# Description
+# ===========
+#
+# We want to build a function :math:`f : \mathbb{R}^d \mapsto \mathbb{R}` from *p* functions :math:`f_i: \mathbb{R} \mapsto \mathbb{R}`.
+#
+# We can do that:
+#
+# - Case 1: by tensorisation of the functions :math:`f_i`,
+# - Case 2: by addition of the functions :math:`f_i`.
+#
+# We need to implement both basic steps:
+#
+# - the projection function: :math:`\pi_i : (x_1, \dots, x_d) \mapsto x_i`,
+# - the composed function: :math:`g_i = f_i \circ \pi_i : (x_1, \dots, x_d) \mapsto f_i(x_i)`.
+
+
+# %%
+# Step 1: Creation of the  projection function
+# ============================================
+#
+# The projection function is defined by:
+#
+# .. math::
+#
+#      \pi_i : (x_1, \dots, x_d) \mapsto x_i
+#
+# We can do that using :
+#
+# - a :class:`~openturns.SymbolicFunction` class,
+# - a :class:`~openturns.LinearFunction` class.
+
+# %%
+# Method 1: We use the :class:`~openturns.SymbolicFunction` class.
+
+
+def buidProjSymbolic(p, i):
+    # R^p --> R
+    # (x1, ..., xp) --> xi
+    inputVar = ot.Description.BuildDefault(p, 'x')
+    return ot.SymbolicFunction(inputVar, [inputVar[i]])
+
+
+d = 2
+all_projections = [buidProjSymbolic(d, i) for i in range(d)]
+print(all_projections[0].getInputDimension(), all_projections[0].getOutputDimension())
+
+# %%
+# Method 2: We use the :class:`~openturns.LinearFunction` class.
+#
+# The function :math:`\pi_i(\vect{x}) = \mat{A}\left(\vect{x}-\vect{c}\right) + \vect{b}`.
+
+
+def buildProjLinear(d, i):
+    # R^d --> R
+    # (x1, ..., xd) --> xi
+    matA = ot.Matrix(1, d)
+    matA[0, i] = 1.0
+    cVect = [0.0] * d
+    bVect = [0.0]
+    return ot.LinearFunction(cVect, bVect, matA)
+
+
+all_projections = [buildProjLinear(d, i) for i in range(d)]
+
+# %%
+# Step 2: Creation of the  composed function
+# ==========================================
+#
+# The composed function is defined by: :math:`g_i = f_i \circ \pi_i` defined by:
+#
+# .. math::
+#
+#      g_i: (x_1, \dots, x_d) \mapsto f_i(x_i)
+#
+# We use the :class:`~openturns.ComposedFunction` class.
+
+f1 = ot.SymbolicFunction(['x1'], ['x1^2'])
+f2 = ot.SymbolicFunction(['x2'], ['3*x2'])
+fi_list = [f1, f2]
+all_g = [ot.ComposedFunction(f, proj) for (f, proj) in zip(fi_list, all_projections)]
+print(all_g[0].getInputDimension(), all_g[0].getOutputDimension())
+
+# %%
+# Case 1: Tensorization
+# =====================
+#
+# We want to build the function :math:`f : \mathbb{R}^d \mapsto \mathbb{R}` defined by:
+#
+# .. math::
+#
+#     f: (x_1, \dots, x_d) \mapsto \prod_{i=1}^d f_i(x_i)
+
+
+def tensorProduct(factors):
+    prod = factors[0]
+    for i in range(1, len(factors)):
+        prod *= factors[i]
```

Comment:
This can be simplified, so that bugs can be avoided (I would rather avoid vectorization here).
```suggestion
    prod = 1.0
    for i in range(len(factors)):
        prod *= factors[i]
```

Reply:
Your suggestion does not work because OT can multiply a function by another one only . In your code, you multiply the scalar 1 by factor[i] which is a function... That is the reason why I initialized the variable prod with a function.